### PR TITLE
[US041] Add Authorisation System To UI

### PR DESCRIPTION
**Pull request by TaylorJCrawford**

### CODE REVIEW

**Review by glup100 — APPROVED** [review]
Looks good to me, nice work

File: controller/AuthController.ts [hunk 1]
```diff
@@ -18,14 +18,14 @@ module.exports = function(app: Application) {
 
             res.redirect('/');
         } catch(e) {
-            res.locals.errormessage = e.message;
+            console.log(e.message);
+            res.locals.errormessage = e.message.message;
```
Comment (glenn-hamilton) [inline controller/AuthController.ts]: `e.message.message`

Comment (TaylorJCrawford) [reply controller/AuthController.ts]: Fixed :)

File: controller/HomeController.ts [hunk 2]
```diff
@@ -5,7 +5,14 @@ module.exports = function(app: Application) {
 
     app.get('/', async (req: Request, res: Response) => {
 
+        console.log(req.header);
```
Comment (glenn-hamilton) [inline controller/HomeController.ts]: Don't need to log this

File: service/HelloWorldService.ts
Comment (glenn-hamilton) [inline service/HelloWorldService.ts:1]: Is it perhaps time to get rid of `Hello World`?

File: views/_includes/layouts/base.html [hunk 3]
```diff
@@ -36,10 +36,20 @@
                         </ul>
                         <div class="right-aligned">
                             {% if (sessionValid) %}
-                            <a class="btn btn-outline-success me-2" href="/logout">
+                            <!-- {TO DO} NEED TO UDPATE TO POST. -->
```
Comment (glenn-hamilton) [inline views/_includes/layouts/base.html]: Is this comment still needed?

File: views/_includes/layouts/base.html [hunk 4]
```diff
@@ -36,10 +36,20 @@
...
+                                <a class="btn btn-outline-success me-2" style="cursor: pointer;"
+                                onclick="this.parentNode.submit();">
                                 <i class="bi bi-box-arrow-right"></i>
                                 <h3 class="hidden">Logout</h3>
                             </a>
+                              </form>
+                            <!-- <a class="btn btn-outline-success me-2" href="/logout">
+                                <i class="bi bi-box-arrow-right"></i>
+                                <h3 class="hidden">Logout</h3>
+                            </a> -->
```
Comment (glenn-hamilton) [inline views/_includes/layouts/base.html]: Is this commented code still needed?

**Review by glenn-hamilton — APPROVED** [review]

File: views/_includes/layouts/base.html (hunk 4 above)
Comment (TaylorJCrawford) [reply views/_includes/layouts/base.html]: Fixed :)

File: views/_includes/layouts/base.html (hunk 3 above)
Comment (TaylorJCrawford) [reply views/_includes/layouts/base.html]: Fixed :) - Nope

File: controller/HomeController.ts (hunk 2 above)
Comment (TaylorJCrawford) [reply controller/HomeController.ts]: Fixed :)

File: service/HelloWorldService.ts
Comment (TaylorJCrawford) [reply service/HelloWorldService.ts:1]: I'll check with the team -> to see if they still want this...

File: controller/AuthController.ts [hunk 5]
```diff
@@ -18,14 +18,14 @@ module.exports = function(app: Application) {
 
             res.redirect('/');
         } catch(e) {
+            console.log(e.message);
```
Comment (glenn-hamilton) [inline controller/AuthController.ts]: 🤔

`console.error()`

Comment (TaylorJCrawford) [reply controller/AuthController.ts]: Yip, fixed now. :)

**Pull request merged**